src/Util/PayLaterAvailabilityChecker.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swag\PayPal\Util;
+
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class PayLaterAvailabilityChecker
```

Comment:
If we evaluate always with an `AvailabilityContext` like I suggested above, we can move this back into the `PayLaterMethodData` and use this as a single point of truth.

Reply:
If I move it back to `PayLaterMethodData` I'm going to have to inject that class into two services, where it has no sense being (`PayPalExpressCheckoutDataService`, `InstallmentBannerSubscriber`). Currently our single source of truth is this class. It holds all the criteria and the logic for whether PayLater is available for both the product and the cart.

Reply:
But my OCD is really triggered that this is now the only method that has such a static class instead of all of them going through the `isAvailable` method of the MethodData. @cyl3x @lernhart any opinions?

Reply:
`MethodData` is not available anywhere. You want to inject it into unrelated classes, JUST so you can use the `isAvailable` method on it? That doesn't make sense to me. 

On top of that you have all the criteria then also in a class that has no use for the criteria. So the class is doing two things now:
1. handling the PayLater payment method
2. checking to make sure that the PayLater payment method, button and banners are not displayed when they shouldn't be.

Reply:
What do you mean `MethodData` is not available anywhere? It's a service like any other.

Your availability checker also needs to be used in the method data, so just merge that class/function into `PayLaterMethodData::isAvailable` so it stays the way it was indented to be used: As a single point of truth for all specifications of this payment method.

Reply:
As stated in the daily, this has been discussed with @mstegmeyer , and instead of going back and forth, I'm moving it to `PayLaterMethodData::isAvailable` even though it's 100% not the correct solution.

Just so you are on the same page. Injecting a full object into a bunch of classes just to solve a non-existent problem that a static class with one method 'creates' isn't the correct way.

The single point of truth argument also holds no water, since the static object was created as the single point of truth.